### PR TITLE
[MIG][10.0] account_bank_statement_import_mt940_base: migration to 10.0

### PR DESCRIPTION
**Pull request by astirpe**
This is the porting of module `account_bank_statement_import_mt940_base` to V10.

Based on the job made in https://github.com/OCA/bank-statement-import/pull/98 by @pedrocasi

### CODE REVIEW

**Comment by melroy89** [comment]
@astirpe Hulp nodig? Ik zou SNS bank & Regiobank kunnen testen..

**Comment by pedrocs-exo** [comment]
Hi, how can I help getting this job done?

**Comment by astirpe** [comment]
Hi, this module for V10 is running in production on the machines of a customer of us.
To test this module we need extra modules, one for each bank format. For example:

- account_bank_statement_import_mt940_nl_rabo
- account_bank_statement_import_mt940_nl_ing

They are available for V8 but I didn't port yet to V10, as our customers do not use them. If someone wants to make a port I will try to help, but for the moment I'm too much busy with other projects.

Instead, regarding modules for SNS bank & Regiobank, not sure whether they are available somewhere. In any case, I don't have any module for such banks. If you know the existence of such modules, they could be used for testing this one and perhaps being proposed for being hosted in OCA.

For porting this module to V10 I created a module parsing the Deutsche bank format. I will ask our customer whether they allow that module to be proposed to OCA.

**Comment by pedrocs-exo** [comment]
Hello,
The module seems to be working just fine. I used it together with a dependent module that I wrote and successfully imported several mt940 statements from the portuguese bank EuroBic. I will submit it to the OCA as soon as your module is commited.

As for the the ING and Rabo bank, I have no interest/time in developing them. I will instead try to identify more banks operating in Portugal that use the format. If I find them and have access to real-world data, I'll try developing specific modules.

**Comment by pedrobaeza** [comment]
Isn't there the possibility of adding tests to the module for not depending on other modules implementation?

**Comment by astirpe** [comment]
@pedrobaeza yes, is possible. I will write some tests to make this module more independent.

**Comment by astirpe** [comment]
According with [this coverage report](https://codecov.io/gh/OCA/bank-statement-import/tree/9a32d27bc3635e8895e84d66724fffe90b2af5ab) this module is 91.97% covered. But the codecov/patch percentage is stuck to 81.81%, less than what I'm expecting. I don't understand why this is happening.

**Comment by pedrobaeza** [comment]
If you see https://codecov.io/gh/OCA/bank-statement-import/compare/18e814d841b0215148c613ccbbd097dc05cefc7b...9a32d27bc3635e8895e84d66724fffe90b2af5ab/diff, you will find lines that are yet not covered, although the general repository coverage has increased. Check if they are exceptions that don't need to be tested, and then put `# pragma: no cover` in them, or expand the tests for that cases. It's also acceptable if you don't want to spend more time. This is far better than previous state.

**Comment by astirpe** [comment]
@pedrobaeza thanks!
Please notice that with latest commit the codecov/project increased (+3%) but the codecov/patch percentage didn't increase, it's still stuck to 81.81%. I noticed this strange behavior also with previous commits.

**Comment by pedrobaeza** [comment]
OK, don't worry more about that and it has been a great effort by your part. Please squash commits at your taste and this one is good to be reviewed.

**Review by feketemihai — APPROVED** [review]

File: account_bank_statement_import_mt940_base/tests/test_import_bank_statement.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
```
Comment (melroy89) [inline account_bank_statement_import_mt940_base/tests/test_import_bank_statement.py:1]: Could you also add a testcase for SNS Bank MT940 files, just to be sure besides Rabobank & ING. See attachment for an example file. 

[tr-info_15785279_20180104123832.zip](https://github.com/OCA/bank-statement-import/files/1603485/tr-info_15785279_20180104123832.zip)

And even documentation if needed:
[6_0016_35_1511_Informatiewijzer_MT940_rapportage.pdf](https://github.com/OCA/bank-statement-import/files/1603487/6_0016_35_1511_Informatiewijzer_MT940_rapportage.pdf)

Ps. I made the same remark on v11 PR: https://github.com/OCA/bank-statement-import/pull/137/files/39e7b8871db5f23bc35013318bc60ba3c8f5b5bb#diff-5bea8377b2408bd073f1a0f6508bd10b

**Review by melroy89 — CHANGES_REQUESTED** [review]
Add SNS Bank MT940 testcase.

**Review by melroy89 — APPROVED** [review]

**Comment by astirpe** [comment]
Can we merge this one?

**Comment by feketemihai** [comment]
@pedrobaeza can you please merge this one..

**Pull request merged**